pyodide-build/pyodide_build/pypabuild.py

```diff
@@ -0,0 +1,108 @@
+import contextlib
+import os
+import sys
+import traceback
+from itertools import chain
+from pathlib import Path
+from typing import Mapping
+
+from build import BuildBackendException, ProjectBuilder  # type: ignore[import]
+from build.__main__ import (  # type: ignore[import]
+    _STYLES,
+    _error,
+    _handle_build_error,
+    _IsolatedEnvBuilder,
+    _ProjectBuilder,
+)
+from build.env import IsolatedEnv  # type: ignore[import]
+from packaging.requirements import Requirement
+
+from .common import get_hostsitepackages, get_pyversion
+
+UNISOLATED_PACKAGES = ["numpy", "scipy", "cffi", "pycparser", "pythran", "cython"]
```

Comment:
So, if I am understanding correctly, one should add a package here if he/she wants to make the package unisolated? I'm not sure whether this list would be updated often, but if so, I think we should provide more easy and non-hardcoded way to update the list (for now it's fine).

Reply:
> one should add a package here if he/she wants to make the package unisolated?

Yes.

> I'm not sure whether this list would be updated often

I hope not.

> think we should provide more easy and non-hardcoded way to update the list

I was thinking of adding a key to `meta.yaml`, but this requires some extra logic to collect the `UNISOLATED_PACKAGES`, so I think we can do that later. Even if we don't change it much, I think that having our build logic care about certain packages by name (like scipy) is a bit surprising. If nothing else, it would probably be a good idea to put a really obvious warning in the `meta.yaml` file.

Reply:
This would be a list of packages with C extensions that show up in build requirements, which is pretty small, but could grow. For something like `cmake`, which is only needed at build time (it's not linked in or anything), I would assume that would not be added here?

Reply:
Anything build time only should be isolated. This includes pythran, which I have unisolated as a hack rather than actually figuring out what is going wrong with scipy's build. For some reason, isolated pythran feature detects that there is no 128 bit float support even though emscripten has perfectly fine 128 bit float support. This breaks the boost stats module build.